### PR TITLE
Fix Connect Edges throwing exception with degenerate tris

### PR DESCRIPTION
**Pull request by karl-**
Fixes fogbugz case 1077016

This also cleans up the functions to remove degenerate triangles and unused vertices, moving them to a new `MeshValidation` class. A new method is added (`MeshValidation.EnsureMeshIsValid`) which cleans up degenerate tris and unused vertices while keeping the selection intact.

### CODE REVIEW

File: com.unity.probuilder/Runtime/MeshOperations/MeshValidation.cs [hunk 1]
```diff
@@ -0,0 +1,306 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace UnityEngine.ProBuilder.MeshOperations
+{
+    /// <summary>
+    /// Methods for testing that mesh topology is correct.
```
Comment (thomas-tu) [inline com.unity.probuilder/Runtime/MeshOperations/MeshValidation.cs]: This class seems to do more than just testing ProBuilderMeshes with the Rebuild/Remove methods. Maybe add a few words more info about.

File: com.unity.probuilder/Editor/EditorCore/RepairActions.cs [hunk 2]
```diff
@@ -139,17 +136,17 @@ public static void MenuRemoveDegenerateTriangles()
 
             foreach (ProBuilderMesh pb in InternalUtility.GetComponents<ProBuilderMesh>(Selection.transforms))
             {
-                pb.ToMesh();
-
-                int[] rm = pb.RemoveDegenerateTriangles();
-                count += rm != null ? rm.Length : 0;
+                var removed = MeshValidation.EnsureMeshIsValid(pb);
```
Comment (neilunity) [inline com.unity.probuilder/Editor/EditorCore/RepairActions.cs]: returned value is not documented properly, change method signature or document correctly

Comment (karl-) [reply com.unity.probuilder/Editor/EditorCore/RepairActions.cs]: 14789d9c

File: com.unity.probuilder/Runtime/MeshOperations/MeshValidation.cs (hunk 1 above)
Comment (karl-) [reply com.unity.probuilder/Runtime/MeshOperations/MeshValidation.cs]: 14789d9c

**Pull request merged**